cdap-ui/app/services/plumber/my-plumb-service.js

```diff
@@ -62,25 +62,37 @@ angular.module(PKG.name + '.services')
         icon: conf.icon,
         description: conf.description,
         type: conf.type
-      }
+      };
       this.nodes[config.id] = config;
       this.notifyListeners(config, type);
     };
 
     this.editPluginProperties = function (scope, pluginId, pluginType) {
-      var plugin = this.nodes[pluginId];
-      var prom = $q.defer();
       var propertiesApiMap = {
         'source': myAdapterApi.fetchSourceProperties,
         'sink': myAdapterApi.fetchSinkProperties,
         'transform': myAdapterApi.fetchTransformProperties
       };
+
+      var sourceConn = $filter('filter')(this.connections, { target: pluginId });
```

Comment:
Will have to write a helper function to do that, because this is filtering on embedded property value. So we need to traverse the entire array and check for that key/value pair.  IndexOf does not have that functionality.

What do you think?